portal/static/js/gil.js

```diff
@@ -545,68 +574,111 @@ module.exports = OrgTool = (function() {
-                $(this).on("click", function() {
+          $("#modal-org").on("hide.bs.modal", function(e) {
+            if (typeof sessionStorage !== "undefined") {
+              sessionStorage.setItem("noOrgModalViewed", "true");
+            } else {
+              alert(i18next.t("Unable to set session variable for organization modal viewed."));
+            }
+            setTimeout(function() { location.reload(); }, 0);
+          });
 
-                    var userId = $("#fillOrgs").attr("userId");
+          $("#submit-orgs").on("click", function() {
+            var os = $("#userOrgs input[name='organization']:checked");
+            if (os.length > 0) {
+              os.each(function() {
+                  if ($(this).attr("id") !== "noOrgs") {
                     var parentOrg = $(this).attr("data-parent-id");
+                    if (!parentOrg) {
+                      parentOrg = $(this).closest(".org-container[data-parent-id]").attr("data-parent-id");
+                    };
+                    var agreementUrl = $("#" + parentOrg + "_agreement_url").val();
+                    if (agreementUrl && String(agreementUrl) !== "") {
+                      var params = CONSENT_ENUM["consented"];
+                      params.org = parentOrg;
+                      params.agreementUrl = agreementUrl;
+                      setTimeout(function() {
+                        self.setConsent($("#currentUserId").val(), params, "all", true);
+                        }, 0);
+                    } else {
+                      self.setDefaultConsent($("#currentUserId").val(), parentOrg);
+                    };
+                  };
+              });
+              self.updateOrg($("#currentUserId").val(), self.updateOrgCallback);
+            };
+          });
 
-                    if ($(this).prop("checked")){
-                        if ($(this).attr("id") !== "noOrgs") {
-                            //console.log("set no org here")
-                            $("#noOrgs").prop('checked',false);
+          $("#userOrgs input[name='organization']").each(function() {
+              $(this).on("click", function() {
 
-                        } else {
-                            $("#userOrgs input[name='organization']").each(function() {
-                                //console.log("in id: " + $(this).attr("id"))
-                               if ($(this).attr("id") !== "noOrgs") {
-                                    $(this).prop('checked',false);
-                                };
-                            });
-
-                        };
-                    };
-                });
-            });
+                  var userId = $("#fillOrgs").attr("userId");
+                  var parentOrg = $(this).attr("data-parent-id");
+
+                  if ($(this).prop("checked")){
+                      if ($(this).attr("id") !== "noOrgs") {
+                          //console.log("set no org here")
```

Comment:
Could you remove this, or have it only log if the log level is high enough (ie debugging on or development system)?